server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -313,7 +313,9 @@ String innerDetect(ThreadMXBean threadBean, SunThreadInfo sunThreadInfo, long cu
                 );
                 case CPU -> {
                     double percentCpu = getTimeSharePercentage(topThread.getCpuTime());
-                    double percentOther = getTimeSharePercentage(topThread.getOtherTime());
+                    double percentOther = Transports.isTransportThread(threadName) && topThread.getCpuTime() == 0L
+                        ? 100.0
```

Comment:
Should this be `100 - topThread.getCpuTime`? Like, is all the stuff no in `cpuTime` idle?

Reply:
Mmm kinda, but (a) the cpu time is zero on this branch so subtracting it makes no difference, and (b) if we made this change on the nonzero-cpu-time branch then we would lose the ability to spot cases where the thread was actually not `RUNNABLE` for some time (e.g. waiting on a mutex, which is not what we consider to be "idle" here).

Reply:
Ah! I see. There's other stuff not in getCpuTime. Makes sense.